File: 백준/Silver/1406. 에디터/에디터.cc
#include <iostream>
#include <stack>
using namespace std;

string s;
int n;
stack<char> st1, st2;

int main() {
    cin >> s;
    cin >> n;

    char command, num;

    for (int i = 0; i < s.length(); i++) {
        st1.push(s[i]);
    }

    while (n--) {
        cin >> command;
        
        if (command == 'L') {
            if (!st1.empty()) {
                st2.push(st1.top());
                st1.pop();
            }
        } else if (command == 'D') {
            if (!st2.empty()) {
                st1.push(st2.top());
                st2.pop();
            }
        } else if (command == 'B') {
            if (!st1.empty()) {
                st1.pop();
            }
        } else if (command == 'P') {
            cin >> num;
            st1.push(num);
        }
    }

    // st1에 있는 내용 출력
    while (!st1.empty()) {
        st2.push(st1.top());
        st1.pop();
    }

    // 결과 출력
    while (!st2.empty()) {
        cout << st2.top();
        st2.pop();
    }

    return 0;
}

File: 백준/Silver/1406. 에디터/에디터.py
import sys
input = sys.stdin.readline

st1 = list(input().rstrip())
n = int(input())

st2 = []

for _ in range(n):
    command = input().split()
    
    if command[0] == 'L':
        if st1:
            st2.append(st1.pop())

    elif command[0] == 'D':
        if st2:
            st1.append(st2.pop())

    elif command[0] == 'B':
        if st1:
            st1.pop()

    elif command[0] == 'P':
        st1.append(command[1])
    
st1.extend(reversed(st2))
print(''.join(st1))